src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/Orleans3CompatibleStringKeyHasher.cs

```diff
@@ -42,24 +42,34 @@ public int Hash(byte[] data)
                 return _innerHasher.Hash(data);
 
             var extendedLength = data.Length + 8;
-            if (extendedLength <= 256)
-            {
-                Span<byte> extended = stackalloc byte[extendedLength];
-                data.AsSpan().CopyTo(extended);
-                return _innerHasher.Hash(extended);
-            }
 
-            var buffer = ArrayPool<byte>.Shared.Rent(extendedLength);
-            try
+            var buffer = extendedLength switch
```

Comment:
This introduces branches that are not necessary. Keep it simple like
```c#
int extendedLenght = data.Length + 8;
byte[]? bufferFromPool = null;

Span<byte> buffer = (extendedLenght <= 256
    ? stackalloc byte[256]
    : ArrayPool<byte>.Shared.Rent(extendedLenght)
).Slice(0, extendedLenght);

data.AsSpan().CopyTo(buffer);
buffer.Slice(data.Length).Clear();

// Hash here

if (bufferFromPool is not null)
{
    ArrayPool<byte>.Shared.Return(bufferFromPool);
}
```

Reply:
but with one more `if` to assign `bufferFromPool`

Reply:
in code above `bufferFromPool` is never assigned

Reply:
Ah sorry, this is from typing the code here within the comments. It's corrected now.